PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/ResourceGroupPanel.java

```diff
@@ -0,0 +1,182 @@
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.runner.functions.component;
+
+import com.intellij.ui.ListCellRendererWrapper;
+import com.intellij.ui.PopupMenuListenerAdapter;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azuretools.core.mvp.model.AzureMvpModel;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.event.PopupMenuEvent;
+import java.awt.Window;
+import java.awt.event.ItemListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.List;
+
+public class ResourceGroupPanel extends JPanel {
+    public static final String CREATE_RESOURCE_GROUP = "Create resource group";
+    private JComboBox cbResourceGroup;
+    private JPanel panel1;
+
+    private Window window;
+    private String subscriptionId;
+    private ResourceGroupWrapper selectedResourceGroup;
+
+    public ResourceGroupPanel(Window window) {
+        this();
+        this.window = window;
+    }
+
+    public ResourceGroupPanel() {
+        cbResourceGroup.setRenderer(new ListCellRendererWrapper() {
+            @Override
+            public void customize(JList list, Object object, int i, boolean b, boolean b1) {
+                if (object instanceof ResourceGroupWrapper) {
+                    setText(((ResourceGroupWrapper) object).getDisplayName());
+                } else if (object instanceof String) {
+                    setText(object.toString());
+                }
+            }
+        });
+
+        // Set label for screen reader for accessibility issue
+        final JLabel label = new JLabel("Resource Group");
+        label.setLabelFor(cbResourceGroup);
+
+        cbResourceGroup.addPopupMenuListener(new PopupMenuListenerAdapter() {
+            @Override
+            public void popupMenuWillBecomeInvisible(PopupMenuEvent e) {
+                onSelectResourceGroup();
+            }
+        });
+    }
+
+    private void createResourceGroup() {
+        cbResourceGroup.setSelectedItem(null);
+        final NewResourceGroupDialog dialog = new NewResourceGroupDialog();
+        dialog.pack();
+        dialog.setLocationRelativeTo(this);
+        dialog.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosed(WindowEvent windowEvent) {
+                super.windowClosed(windowEvent);
+                final ResourceGroupWrapper newResourceGroup = dialog.getResourceGroup();
```

Comment:
where is the validation part?

Reply:
Validation will be submitted in later PR